### ISSUE TITLE
MR_newContext creates context with root saving context parent.

### ISSUE THREAD

**Issue by albertodebortoli** [magicalpanda/MagicalRecord#1059]
The documentation states

+ [NSManagedObjectContext MR_newContext]: Sets the default context as it's parent context. Has a concurrency type of NSPrivateQueueConcurrencyType.

https://github.com/magicalpanda/MagicalRecord/blob/master/Docs/Working-with-Managed-Object-Contexts.md

but the implementation returns [self MR_contextWithParent:[self MR_rootSavingContext]];

This is particularly bad as the default context is not updated and against well-known designs (e.g. http://martiancraft.com/blog/2015/03/core-data-stack/).

**Comment by zdesiree** [comment]
I use the method `let newContext = NSManagedObjectContext.MR_context()` to create a new context and sometimes in the production app I get a crash that says:
```
Fatal Exception: NSInvalidArgumentException
Can only use -performBlockAndWait: on an NSManagedObjectContext that was created with a queue.
```

Could this be a related issue?

**Comment by tonyarnold** [comment]
@zdesiree no, that's a separate issue. `MR_context()` returns a context initialised using a thread confinement model in the current release, which cannot use `performBlock:` and `performBlockAndWait:`. I'm working on weeding out all of our uses of thread confinement, as it has been deprecated in iOS 9 / OS X 10.11, but that's going to take some time.
